src/SixLabors.Fonts/Tables/AdvancedTypographic/GPos/LookupType6SubTable.cs

```diff
@@ -0,0 +1,132 @@
+                // |                    |                                 | from beginning of MarkMarkPos subtable.              |
+                // +--------------------+---------------------------------+------------------------------------------------------+
+                // | Offset16           | mark2ArrayOffset                | Offset to Mark2Array table for mark2,                |
+                // |                    |                                 | from beginning of MarkMarkPos subtable.              |
+                // +--------------------+---------------------------------+------------------------------------------------------+
+                ushort mark1CoverageOffset = reader.ReadOffset16();
+                ushort mark2CoverageOffset = reader.ReadOffset16();
+                ushort markClassCount = reader.ReadUInt16();
+                ushort mark1ArrayOffset = reader.ReadOffset16();
+                ushort mark2ArrayOffset = reader.ReadOffset16();
+
+                var mark1Coverage = CoverageTable.Load(reader, offset + mark1CoverageOffset);
+                var mark2Coverage = CoverageTable.Load(reader, offset + mark2CoverageOffset);
+                var mark1ArrayTable = new MarkArrayTable(reader, offset + mark1ArrayOffset);
+                var mark2ArrayTable = new Mark2ArrayTable(reader, markClassCount, offset + mark2ArrayOffset);
+
+                return new LookupType6Format1SubTable(mark1Coverage, mark2Coverage, mark1ArrayTable, mark2ArrayTable);
+            }
+
+            public override bool TryUpdatePosition(
+                IFontMetrics fontMetrics,
+                GPosTable table,
+                GlyphPositioningCollection collection,
+                Tag feature,
+                ushort index,
+                int count)
+            {
+                // Mark to mark positioning.
+                // Implements: https://docs.microsoft.com/en-us/typography/opentype/spec/gpos#lookup-type-6-mark-to-mark-attachment-positioning-subtable
+                ushort glyphId = collection[index][0];
+                if (glyphId == 0)
+                {
+                    return false;
+                }
+
+                int mark1Index = this.mark1Coverage.CoverageIndexOf(glyphId);
+                if (mark1Index == -1)
+                {
+                    return false;
+                }
+
+                // Get the previous mark to attach to.
+                if (index < 1)
+                {
+                    return false;
+                }
+
+                // TODO: fontkit checks here, if Marks belonging to the same base or Marks belonging to the same ligature component.
+                int prevIdx = index - 1;
+                ushort prevGlyphId = collection[prevIdx][0];
+                GlyphShapingData prevData = collection.GetGlyphShapingData(prevIdx);
+                if (!CodePoint.IsMark(prevData.CodePoint))
+                {
+                    return false;
```

Comment:
`\0012` should be a Mark in the font GposLookupType6Format1.ttf, but `CodePoint.IsMark` returns false here. Maybe checking if its a mark via CodePoint is not correct here.

Reply:
Cool. I’ll have a look tonight for you

Reply:
Looking now.

Reply:
if you load the font with Font Creator, you can see the `OT Class` of `\0012` to be Mark.

Reply:
Ok. So our `CodePoint.IsMark()` check is correct, however this is not what FontKit nor HarfBuzz are checking. They both maintain `LigatureId` and `LigatureComponent` properties for each glyph and check those. 

We need to add those properties to the `GlyphShapingData` type and also a `SetGlyphShapingData(index, data)` method to `GlyphPositioningCollection`. We'll have to ensure we look back through the FontKit source to see when those properties are ever set.

Reply:
We very likely need GDEF table support.
https://docs.microsoft.com/en-us/typography/opentype/spec/gdef

Reply:
Hold on..... Yep. It's a combining mark when you check its LineBreakClass..... I need to do some research to make the IsMark test more robust!

Reply:
Ok So I did a whole bunch of research and augmented the method with the following code to cover all available ranges however that then broke lookups 4 and 5.

https://en.wikipedia.org/wiki/Combining_character
https://unicodesymbols.fandom.com/wiki/Category:Combining_Mark_Set

```c#
public static bool IsMark(CodePoint codePoint)
{
    // https://en.wikipedia.org/wiki/Combining_character

    // Combining Diacritical Marks
    if (UnicodeUtility.IsInRangeInclusive(codePoint.value, 0x0300, 0x036F))
    {
        return true;
    }

    // Combining Diacritical Marks Extended
    if (UnicodeUtility.IsInRangeInclusive(codePoint.value, 0x1AB0, 0x1AFF))
    {
        return true;
    }

    // Combining Diacritical Marks Supplement
    if (UnicodeUtility.IsInRangeInclusive(codePoint.value, 0x1DC0, 0x1DFF))
    {
        return true;
    }

    // Combining Diacritical Marks for Symbols
    if (UnicodeUtility.IsInRangeInclusive(codePoint.value, 0x20D0, 0x20FF))
    {
        return true;
    }

    // Combining Half Marks
    if (UnicodeUtility.IsInRangeInclusive(codePoint.value, 0xFE20, 0xFE2F))
    {
        return true;
    }

    return IsCategoryMark(GetGeneralCategory(codePoint)) || GetLineBreakClass(codePoint) == LineBreakClass.CM;
}
```

I'm not sure what the next step is. Do we simply check the linebreak class in lookup 6?

Reply:
My understanding so far was, that we need to implement the GDEF table and get the info, if the glyph is a mark from there, right?

Reply:
Unfortunately so. Unicode support should be enough but it doesn’t seem to be.

Reply:
Fontkit cheats by storing multiple codepoints against a glyph following a removal and counting that as a mark. Theoretically we could do the same but it’s yet another array

Reply:
Ok, I will start then implementing the GDEF table

Reply:
so far it seems to work out as planed: I use now the GDEF table if present to get the glyph class otherwise CodePoint.IsMark in LookupType6SubTable

Reply:
Awesome! That was quick!